six/common/builtins/aggregator.c

```diff
@@ -83,7 +83,10 @@ static char **py_tag_to_c(PyObject *py_tags)
     }
 
     int len = PySequence_Length(py_tags);
-    if (len == 0) {
+    if (len == -1) {
+        PyErr_SetString(PyExc_RuntimeError, "could not compute tags length");
```

Comment:
is it better to set a new error with `PyErr_SetString` here, or simply keep the context of the original error by leaving the error as-is and returning `NULL`?

Reply:
Which error are you refering to ? From the doc `PySequence_Length` will not raise anything.

Reply:
The [doc](https://docs.python.org/2/c-api/sequence.html#c.PySequence_Length) mentions that `PySequence_Length` returns `-1` on failure, and that generally means that an error is set on the interpreter when it does. This is confirmed by the code: https://github.com/python/cpython/blob/74bede0d503d0508e9590f4ad65ee48b0ab93db6/Objects/abstract.c#L1859-L1867 (although I'm not sure any of these errors can actually be raised in our code here since we've already checked the object is actually a sequence)

In any case, in the general case I think keeping the original error (or fetching it to add some context) is better, but in this specific case we can merge as-is given how unlikely an error would be.